src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs

```diff
@@ -32,74 +30,72 @@ internal sealed class GraphNavigatorExtension(
 
     public void NavigateTo(GraphObject graphObject)
     {
+        if (graphObject is not GraphNode graphNode)
+            return;
 
-        if (graphObject is GraphNode graphNode)
-        {
-            var sourceLocation = graphNode.GetValue<SourceLocation>(CodeNodeProperties.SourceLocation);
-            if (sourceLocation.FileName == null)
-            {
-                return;
-            }
+        _threadingContext.JoinableTaskFactory.Run(() => NavigateToAsync(graphNode, CancellationToken.None));
+    }
 
-            var projectId = graphNode.GetValue<ProjectId>(RoslynGraphProperties.ContextProjectId);
-            var symbolId = graphNode.GetValue<SymbolKey?>(RoslynGraphProperties.SymbolId);
+    private async Task NavigateToAsync(GraphNode graphNode, CancellationToken cancellationToken)
+    {
+        var projectId = graphNode.GetValue<ProjectId>(RoslynGraphProperties.ContextProjectId);
```

Comment:
Are there any UI thread requirements for this?

Reply:
nope!